src/build/command.rs

```diff
@@ -4008,15 +4008,19 @@ impl<'help> App<'help> {
     /// Call this on the top-level [`Command`] when done building and before reading state for
     /// cases like completions, custom help output, etc.
     pub fn build(&mut self) {
+        self._build_recursive();
+        self._build_bin_names_internal();
+    }
+
+    pub(crate) fn _build_recursive(&mut self) {
         self._build_self();
         for subcmd in self.get_subcommands_mut() {
-            subcmd._build_self();
+            subcmd._build_recursive();
```

Comment:
AFAIK The reason for not building everything was performance concerns. We only built the subcommands people were using.

Reply:
There are two build paths.  The incremental build path used by parsing and the "build all" build path used by completions, etc.  They share `_build_self` but how they walk and how they generate bin names is separate.